src/Microsoft.VisualStudio.LanguageServices.Razor/ProjectSystem/GeneratedDocumentTextLoader.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.Razor.ProjectSystem
+{
+    internal class GeneratedDocumentTextLoader : TextLoader
+    {
+        private readonly DocumentSnapshot _document;
+        private readonly string _filePath;
+        private readonly VersionStamp _version;
+
+        public GeneratedDocumentTextLoader(DocumentSnapshot document, string filePath)
+        {
+            if (document == null)
+            {
+                throw new ArgumentNullException(nameof(document));
+            }
+
+            _document = document;
+            _filePath = filePath;
+            _version = VersionStamp.Create();
```

Comment:
Shouldn't this be looking at the documents version and creating a version that's newer?

Reply:
No, that version isn't meaningful to Roslyn. They also don't use `VersionStamp` to indicate *newer*, they only use `VersionStamp` to indicate *different*